### PR TITLE
Add `CODEOWNER` for the `array_api` module

### LINKED ISSUE #21399: Array-API: Need a bot to notify downstream maintainers for any changes in `numpy.array_api`
Sorry for delay. I was tasked to open this issue based on the discussion in an array API meeting a few weeks ago. We need to set up a bot to detect if there's any diff in the `numpy.array_api` module appearing in a PR, if so the bot should notify a list of people (CuPy, MXNet, ...) upon merge to keep them informed.

cc: @rgommers @asmeurer @kmaehashi @honno

### PR DESCRIPTION
**Pull request by leofang**
Close #21399. 

Question: Who else should be listed here?

cc: @kmaehashi @rgommers @asmeurer @honno @seberg

### CODE REVIEW

**Comment by leofang** [comment]
Hmmm the GUI shows an error saying I need to have write access to this repo, but the GH instruction says individual users only need read access.

**Comment by asmeurer** [comment]
You can add me.

**Comment by kmaehashi** [comment]
I would also like to be added, but CODEOWNERS seems not working in this case 😢

> Does anyone have experience in this?

I noticed I'm making the same mistake in the project that I maintain (not all members specified in [CODEOWNERS](https://github.com/pfnet/pytorch-pfn-extras/blob/master/.github/CODEOWNERS) are notified in [pull-requests](https://github.com/pfnet/pytorch-pfn-extras/pull/539)). GitHub sends notifications by assigning CODEOWNERs to reviewers of pull-requests, but only users who have ["Read" role](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#repository-roles-for-organizations) in the organization can be assigned as a reviewer. This is not intuitive, but the default role of users outside of the organization seems "null" instead of "read".

**Comment by rgommers** [comment]
I can make a new team later today for read-only rights for whoever needs those.

**Comment by leofang** [comment]
Got it, thanks @rgommers!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# See GitHub docs at https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
+
+# For the array API module:
+# We want to notify a group of downstream maintainers about any changes
+# made to numpy.array_api; they are not obligated to perform the code
+# review (but they are more than welcome to help), this merely serves
+# as a one-way notification.
+/numpy/array_api/  @leofang @asmeurer @kmaehashi
```
Comment (honno) [inline .github/CODEOWNERS:8]: Yep I'd like notifications, assuming that's okay @rgommers (I'd need to be invited for `readonly-access`).

Comment (leofang) [reply .github/CODEOWNERS]: Thanks Matthew, done.

**Comment by rgommers** [comment]
Everyone has readonly access now, but GitHub still seems unhappy ...

**Comment by leofang** [comment]
The point is PR authors don't know when to ping the team so there could be misses. I feel this is a bug on GitHub side as they promised code owners only need to have read access. I wonder what would happen if we merge it?

**Comment by seberg** [comment]
Ohhh, so a team that is owner must have write access, but users do not?  But we are listing you as users, and you should have readonly access now?

I agree, that all of this would be good things to have github features for (in whatever style exactly), but it is not like we can hope for that to happen any time soon...

**Comment by leofang** [comment]
I think currently we're members of the NumPy org, but not the NumPy repo, maybe this is the difference? The easiest way to check is to add me as user here, and then look at this PR and see if it still complains 🙂

**Comment by seberg** [comment]
Well, I upgraded the team to triage rights (because why not), and added you as a single user with read access.  Doesn't seem like either helped, unless it is just the validation that is wrong?  I guess could just try merge and try if you like.

**Comment by leofang** [comment]
I confirm I received the notification that access is granted, but nothing seems to change indeed... Let's merge and let me create a dummy PR touching `numpy.array_api` to see what happens?

**Comment by leofang** [comment]
@seberg @rgommers Looks like it didn't work... https://github.com/numpy/numpy/pull/21650 Shall we revert the change?

**Comment by asmeurer** [comment]
By the way, I've heard that GitHub is going to add the ability to subscribe to labels on issues/PRs (no idea when, but apparently some other repos have it as a beta feature). Once that lands, it should hopefully make this situation much simpler, as it would just depend on PRs being labelled properly (which itself can be done with a simple GitHub Action).

**Comment by seberg** [comment]
Yes, please revert.  Lets keep codeowners clear for when it is actually used in a useful way ;)

**Comment by leofang** [comment]
That would be nice... Apparently CODEOWNERS is known to not work for a while...
- https://github.com/github-community/community/discussions/8362
- https://github.community/t/codeowners-needs-more-than-read-permission/250283/2

**Comment by leofang** [comment]
See #21651.

**Pull request merged**